xarray/core/dataset.py

```diff
@@ -1657,6 +1657,45 @@ def broadcast_equals(self, other: Dataset) -> bool:
         the other dataset can still be broadcast equal if the the non-scalar
         variable is a constant.
 
+        Examples
```

Comment:
This is great.

Could improve by taking inspiration from `DataArray.broadcast_equals` docstring (and fixing any problems you find there. For example calling both `.equals` and `.broadcast_equals` is useful. Also call `.broadcast_equals` before `xr.broadcast` to show that it returns False!